gramps/gen/lib/grampstype.py

```diff
@@ -151,11 +158,18 @@ def set_object_state(self, attr_dict):
         else:
             self.__string = ""
 
-    def __set_dict(self, value):
+    @singledispatchmethod
+    def set(self, value):
+        self.__value = self._DEFAULT
+        self.__string = ""
+
+    @set.register
+    def set_dict(self, value: dict):
         "Set the value/string properties from a dict."
         self.__set_tuple((value["value"], value["string"]))
 
-    def __set_tuple(self, value):
+    @set.register
+    def set_tuple(self, value: tuple):
```

Comment:
This should be private.

Reply:
This has been addressed in the latest commit